File: examples/C/hello_world/Windows/VS2019/hello_world/hello_world.c
/*
	Webinix Library 2.x
	C99 Visual Studio Example

	http://webinix.me
	https://github.com/alifcommunity/webinix

	Licensed under GNU General Public License v3.0.
	Copyright (C)2022 Hassan DRAGA <https://github.com/hassandraga>.
*/

#include "webinix.h"

// Window struct
webinix_window_t* my_window;

// UI HTML
const char* my_html = "<!DOCTYPE html>"
"<html><head><title>Webinix 2 - C99 Visual Studio Example</title>"
"<style>body{color: white; background: #0F2027;"
"background: -webkit-linear-gradient(to right, #8d4887, #521b4e, #3e073a);"
"background: linear-gradient(to right, #8d4887, #521b4e, #3e073a);"
"text-align:center; font-size: 18px; font-family: sans-serif;}</style></head><body>"
"<h1>Webinix 2 - C99 Visual Studio Example</h1><br>"
"<input type=\"password\" id=\"MyInput\"><br><br>"
"<button id=\"MyButton1\">Check Password</button> - <button id=\"MyButton2\">Exit</button>"
"</body></html>";

// Check the password function
void check_the_password(webinix_event_t* e) {

	// This function get called every time the user click on "MyButton1"

	webinix_script_t js = {
		.script = " return document.getElementById(\"MyInput\").value; ",
		.timeout = 3
	};

	// Run the JavaScript on the UI (Web Browser)
	webinix_script(my_window, &js);

	// Check if there is any JavaScript error
	if (js.result.error) {

		printf("JavaScript Error: %s\n", js.result.data);
		return;
	}

	// Get the password
	const char* password = js.result.data;
	printf("Password: %s\n", password);

	// Check the password
	if (strcmp(password, "123456") == 0) {

		// Correct password
		js.script = "alert('Good. Password is correct.')";
		webinix_script(my_window, &js);
	}
	else {

		// Wrong password
		js.script = "alert('Sorry. Wrong password.')";
		webinix_script(my_window, &js);
	}

	// Free data resources
	webinix_free_script(&js);
}

void close_the_application(webinix_event_t* e) {

	// Close all opened windows
	webinix_exit();
}

int main() {

	// Create a window
	my_window = webinix_new_window();

	// Bind HTML elements with functions
	webinix_bind(my_window, "MyButton1", check_the_password);
	webinix_bind(my_window, "MyButton2", close_the_application);

	// Show the window
	if (!webinix_show(my_window, my_html, webinix.browser.chrome))	// Run the window on Chrome
		webinix_show(my_window, my_html, webinix.browser.any);		// If not, run on any other installed web browser

	// Wait until all windows get closed
	webinix_wait();

	return 0;
}

int APIENTRY WinMain(HINSTANCE hInst, HINSTANCE hInstPrev, PSTR cmdline, int cmdshow) {

	main();
}

File: examples/TypeScript/Deno/hello_world/hello_world.ts

// Run this script by:
// deno run --allow-all --unstable hello_world.ts

// Import Webinix module
import * as webinix from "../module/webinix.ts";

// Optional - Set a custom library path:
//  const lib_full_path = '../../../../build/Windows/MSVC/webinix-2-x64.dll';
//  console.log("Looking for the Webinix dynamic library at: " + lib_full_path);
//  webinix.set_lib_path(lib_full_path);

const my_html = `
<!DOCTYPE html>
<html>
	<head>
		<title>Webinix 2 - Deno Hello World Example</title>
        <style>
            body {
                color: white;
                background: #101314;
                background: linear-gradient(to right, #101314, #0f1c20, #061f2a);
                text-align: center;
                font-size: 16px;
                font-family: sans-serif;
            }
        </style>
    </head>
    <body>
        <h1>Webinix 2 - Deno Hello World Example</h1><br>
        A: <input id="MyInputA" value="4"><br><br>
        B: <input id="MyInputB" value="6"><br><br>
        <div id="Result" style="color: #dbdd52">A + B = ?</div><br><br>
	    <button id="Calculate">Calculate</button> - <button id="Exit">Exit</button>
        <script>
            function get_A() {
                return parseInt(document.getElementById('MyInputA').value);
            }

            function get_B() {
                return parseInt(document.getElementById('MyInputB').value);
            }

            function set_result(res) {
                document.getElementById("Result").innerHTML = 'A + B = ' + res;
            }
        </script>
    </body>
</html>
`;

function calculate(e : webinix.event) {

    // Create a JS struct
    const MyJS: webinix.js = {
        Script: "",
        Timeout: 0,
    };
 
    // Call JS function `get_A()`
    MyJS.Script = "return get_A();";
    webinix.run(e.win, MyJS);
    const A = MyJS.Data;

    // Check for error
    if(MyJS.Error) {
        console.log("Error in the JavaScript: " + MyJS.Data);
    }

    // Call JS function `get_B()`
    MyJS.Script = "return get_B();";
    webinix.run(e.win, MyJS);
    const B = MyJS.Data;

    const C : Number = parseInt(A) + parseInt(B);

    // Set the result
    MyJS.Script = "set_result(" + C + ");";
    webinix.run(e.win, MyJS);
}

// Create new window
const my_window = webinix.new_window();

// Bind
webinix.bind(my_window, "Calculate", calculate);
webinix.bind(my_window, "Exit", function(e : webinix.event) {
    // Close all windows and exit
    webinix.exit();
});

// Show the window
webinix.show(my_window, my_html);

// Wait until all windows get closed
await webinix.wait();

console.log("Thank you.");
Deno.exit(0);
